repose-aggregator/core/repose-core-api/src/test/scala/org/openrepose/core/container/config/ContainerSchemaTest.scala

```diff
@@ -166,55 +212,102 @@ class ContainerSchemaTest extends FunSpec with Matchers {
           |        <artifact-directory>/usr/share/repose/filters</artifact-directory>
-          |        <ssl-configuration>
-          |            <keystore-filename>keystore.jks</keystore-filename>
-          |            <keystore-password>password</keystore-password>
-          |            <key-password>password</key-password>
-          |        </ssl-configuration>
           |    </deployment-config>
           |
           |    <cluster-config cluster-id="foo">
-          |        <ssl-configuration>
-          |            <key-password>other-password</key-password>
-          |        </ssl-configuration>
+          |        <via-configuration enabled="false"/>
           |    </cluster-config>
           |</repose-container>""".stripMargin
       validator.validateConfigString(config)
     }
   }
+
+  private val coreSpringProvider = CoreSpringProvider.getInstance()
+  coreSpringProvider.initializeCoreContext("/etc/repose", false)
+  private val reposeVersion = coreSpringProvider.getCoreContext.getEnvironment.getProperty(
+    ReposeSpringProperties.stripSpringValueStupidity(ReposeSpringProperties.CORE.REPOSE_VERSION))
+  describe("deprecated schema validation") {
+    if (reposeVersion.startsWith("8.")) {
+      it("should reject a config where cluster IDs are not unique across all patches") {
+        val config =
+          """<repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
+            |    <deployment-config>
+            |        <deployment-directory>/var/repose</deployment-directory>
+            |        <artifact-directory>/usr/share/repose/filters</artifact-directory>
+            |    </deployment-config>
+            |
+            |    <cluster-config cluster-id="foo" via="test"/>
+            |
+            |    <cluster-config cluster-id="foo" via="test"/>
+            |</repose-container>""".stripMargin
+        val exception = intercept[SAXParseException] {
+          validator.validateConfigString(config)
+        }
+        exception.getLocalizedMessage should include("When defining per-cluster configuration, all cluster IDs must be unique")
+      }
+
+      it("should accept a config where cluster IDs are unique across all patches") {
+        val config =
+          """<repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
+            |    <deployment-config>
+            |        <deployment-directory>/var/repose</deployment-directory>
+            |        <artifact-directory>/usr/share/repose/filters</artifact-directory>
+            |    </deployment-config>
+            |
+            |    <cluster-config cluster-id="foo" via="test"/>
+            |
+            |    <cluster-config cluster-id="bar" via="test"/>
+            |</repose-container>""".stripMargin
+        validator.validateConfigString(config)
+      }
+    } else {
+      it("should reject a config that uses the old via attribute") {
```

Comment:
Of the three, this test seems like the one that is valuable. That said, I would add removing this test to the timebomb test notes as well, since it doesn't have value once we remove the `via` attribute.